src/ppmLogger.cpp

```diff
@@ -13,7 +13,27 @@ PpmLogger::PpmLogger(std::string logname) {
         sinks.push_back(std::make_shared<spdlog::sinks::stdout_sink_mt>());
     }
     setLogger(std::make_shared<spdlog::logger>("log", begin(sinks), end(sinks)));
+
+    // use PPM_LOG_LEVEL environment variable to set the log level
+    std::string logLevelString = getEnvironmentVariable("PPM_LOG_LEVEL");
```

Comment:
If PPM_LOG_LEVEL isn't set, the default value will be used. If it is set to a number or anything else unrecognized, the default will also be used. I've added some comments to a few methods to make this clearer.

Since we read environment variables as strings, even if PPM_LOG_LEVEL is set to a number it will be treated as a string, so we won't run into any issues.